syntax = "proto3";

message MyRequest {
  sint64 aaa = 1;
  oneof arg {
    int64 int64_value = 2;
    string string_value = 3;
    bool bool_value = 4;
  }
  int32 bbb = 5;
  oneof aab {
    int64 int64_value2 = 6;
    string string_value2 = 7;
    bool bool_value2 = 8;
  }
}

message MyReply {
  oneof result {
    int64 int64_result = 1;
    string string_result = 2;
  }
}

service One {
  rpc EchoOneOf(MyRequest) returns (MyReply);
}
